bundles/org.eclipse.packagedrone.utils.rpm/src/org/eclipse/packagedrone/utils/rpm/build/RpmBuilder.java

```diff
@@ -754,9 +787,7 @@ private Path makeTargetFile ( final Path target )
 
     private String makeDefaultFileName ()
     {
-        final StringBuilder sb = new StringBuilder ( RpmLead.toLeadName ( this.name, this.version ) );
-        sb.append ( '-' ).append ( this.architecture ).append ( ".rpm" );
-        return sb.toString ();
+		return getRpmFileNameProvider ().getRpmFileName ( this );
```

Comment:
There should be some kind of handling for `null`. You also could directly use the internal field `this.rpmFileNameProvider` as package drone doesn't internally call getters. But in any case it should throw an exception someone sets this to `null`.